src/gridstack.ts

```diff
@@ -1547,78 +1538,31 @@ export class GridStack {
     }
 
-    const cellHeight = this.opts.cellHeight as number;
-    const cellHeightUnit = this.opts.cellHeightUnit;
-    const prefix = `.${this._styleSheetClass} > .${this.opts.itemClass}`;
-
-    // create one as needed
-    if (!this._styles) {
-      // insert style to parent (instead of 'head' by default) to support WebComponent
-      const styleLocation = this.opts.styleInHead ? undefined : this.el.parentNode as HTMLElement;
-      this._styles = Utils.createStylesheet(this._styleSheetClass, styleLocation, {
-        nonce: this.opts.nonce,
-      });
-      if (!this._styles) return this;
-      this._styles._max = 0;
-
-      // these are done once only
-      Utils.addCSSRule(this._styles, prefix, `height: ${cellHeight}${cellHeightUnit}`);
-      // content margins
-      const top: string = this.opts.marginTop + this.opts.marginUnit;
-      const bottom: string = this.opts.marginBottom + this.opts.marginUnit;
-      const right: string = this.opts.marginRight + this.opts.marginUnit;
-      const left: string = this.opts.marginLeft + this.opts.marginUnit;
-      const content = `${prefix} > .grid-stack-item-content`;
-      const placeholder = `.${this._styleSheetClass} > .grid-stack-placeholder > .placeholder-content`;
-      Utils.addCSSRule(this._styles, content, `top: ${top}; right: ${right}; bottom: ${bottom}; left: ${left};`);
-      Utils.addCSSRule(this._styles, placeholder, `top: ${top}; right: ${right}; bottom: ${bottom}; left: ${left};`);
-      // resize handles offset (to match margin)
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-n`, `top: ${top};`);
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-s`, `bottom: ${bottom}`);
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-ne`, `right: ${right}`);
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-e`, `right: ${right}`);
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-se`, `right: ${right}; bottom: ${bottom}`);
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-nw`, `left: ${left}`);
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-w`, `left: ${left}`);
-      Utils.addCSSRule(this._styles, `${prefix} > .ui-resizable-sw`, `left: ${left}; bottom: ${bottom}`);
-    }
-
-    // now update the height specific fields
-    maxH = maxH || this._styles._max;
-    if (maxH > this._styles._max) {
-      const getHeight = (rows: number): string => (cellHeight * rows) + cellHeightUnit;
-      for (let i = this._styles._max + 1; i <= maxH; i++) { // start at 1
-        Utils.addCSSRule(this._styles, `${prefix}[gs-y="${i}"]`, `top: ${getHeight(i)}`);
-        Utils.addCSSRule(this._styles, `${prefix}[gs-h="${i + 1}"]`, `height: ${getHeight(i + 1)}`); // start at 2
-      }
-      this._styles._max = maxH;
-    }
+    // Set CSS var of cell height
+    this.setVar(this.el.parentElement, "--gs-cell-height", `${this.opts.cellHeight}${this.opts.cellHeightUnit}`);

```

Comment:
can we use this.el instead ? the parent div could contain multiple grids which will break if they have different options. the old stylesheet had unique grid id (old 1578) for style to only apply to them...
this will also be needed for nested grid (parent of a grid is a gridItemContent)